CenterForOpenScience/ember-osf-web
# [ENG-2035] Feature/jQuery to Replace Ember Deprecation for Ticket ENG-2035

- Ticket: https://openscience.atlassian.net/browse/ENG-2035
- Feature flag: n/a

Purpose:

The purpose of these changes was to perform routine maintenance tasks on the ember-osf-web repository and remove the deprecated use of Ember.$() in favor of jQuery instead. This has largely taken the form of utilizing the '$' character imported from the jQuery library for Ember. There were some instances where a call to JQuery instead was preferred. 

Future considerations for this work include rewriting the jQuery statements themselves and reverse-transliterate them back into pure JavaScript. Alternatively, if native code or Ember components exist for commonly used code those may also be used.

Summary of Changes:

The following files were affected during this revision:

-app/dashboard/controller.ts
-app/models/osf-model.ts
-app/services/current-user.ts
-config/optional-features.json
-lib/osf-components/addon/components/file-share-button/component.ts

Side Effects:

We may want to look at removing our dependency on jQuery invocations all together. Unexpected side effects when the Ember community upgrades to version 4 may leave this feature behind potentially resulting in breaking changes.

QA Notes/Test Plan:

1. Login to the OSF as 1) admin 2) non-moderator users
2. Test the Discover/main dashboard page still works
3. Ensure no other breaking changes occurred with the OSF model controller to include the Registries page (a bit vague I know - please feel free to message me on Flowdock)
